exchanges/poloniex/poloniex_wrapper.go

```diff
@@ -589,8 +589,147 @@ func (p *Poloniex) CancelAllOrders(_ *order.Cancel) (order.CancelAllResponse, er
 
 // GetOrderInfo returns order information based on order ID
 func (p *Poloniex) GetOrderInfo(orderID string, pair currency.Pair, assetType asset.Item) (order.Detail, error) {
-	var orderDetail order.Detail
-	return orderDetail, common.ErrNotYetImplemented
+	var orderInfo order.Detail
+	trades, err := p.GetAuthenticatedOrderTrades(orderID)
+	if err == nil {
+		orderInfo.Exchange = p.Name
+		orderInfo.Pair = pair
+		for _, o := range trades {
+			var tradeHistory order.TradeHistory
+			tradeHistory.Exchange = p.Name
+			tradeHistory.Side, err = order.StringToOrderSide(o.Type)
+			if err != nil {
+				return orderInfo, err
+			}
+			tradeHistory.TID = strconv.FormatInt(o.GlobalTradeID, 10)
+			tradeHistory.Timestamp, err = time.Parse(common.SimpleTimeFormat, o.Date)
+			if err != nil {
+				return orderInfo, err
+			}
+			tradeHistory.Price = o.Rate
+			tradeHistory.Amount = o.Amount
+			tradeHistory.Total = o.Total
+			tradeHistory.Fee = o.Fee
+
+			orderInfo.Trades = append(orderInfo.Trades, tradeHistory)
+		}
+	}
+
+	resp, err := p.GetAuthenticatedOrderStatus(orderID)
+	if err != nil {
+		return orderInfo, err
+	}
+
+	for _, v := range resp.Result {
+		switch resultType := v.(type) {
+		case string: // case of error message
+			if len(orderInfo.Trades) > 0 { // on closed orders return trades only
+				return orderInfo, nil
+			}
+			return orderInfo, fmt.Errorf(resultType)
+		case map[string]interface{}: // case of data message
+			orderStatus := v.(map[string]interface{})
+			orderInfo.Exchange = p.Name
+
+			if statusInterface, ok := orderStatus["status"]; ok {
+				switch varType := statusInterface.(type) {
+				case string:
+					orderInfo.Status, _ = order.StringToOrderStatus(varType)
```

Comment:
can we check all error returns

Reply:
On error case StringToOrderStatus return an order.UnknownStatus, so in my opinion we no need to handle it

Reply:
That's a good point. Will resolve.